src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
             pairCount++;
         }
         ReturnErrorOnFailure(pair_iterator.GetStatus());
-
-        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList;
-        attributeValueList = mAVPairs;
+        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList(aVPairs);
         attributeValueList.reduce_size(pairCount);
 
-        writer.Init(serializedBytes);
-        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer));
-        ReturnErrorOnFailure(app::DataModel::Encode(
-            writer, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList),
-            attributeValueList));
-        ReturnErrorOnFailure(writer.EndContainer(outer));
-
-        return CHIP_NO_ERROR;
+        return EncodeAttributeValueList(attributeValueList, serializedBytes);
     }
 
     /// @brief Simulates taking data from nvm and loading it in a command object if the cluster is supported by the endpoint
     /// @param endpoint target endpoint
     /// @param cluster  target cluster
-    /// @param serialisedBytes data to deserialize into EFS
+    /// @param serializedBytes data to deserialize into EFS
     /// @return CHIP_NO_ERROR if Extension Field Set was successfully populated, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not
     /// supported, specific CHIP_ERROR otherwise
-    virtual CHIP_ERROR Deserialize(EndpointId endpoint, ClusterId cluster, const ByteSpan & serialisedBytes,
+    virtual CHIP_ERROR Deserialize(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
                                    app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet) override
     {
         app::DataModel::DecodableList<app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
-        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType decodePair;
 
-        TLV::TLVReader reader;
-        TLV::TLVType outer;
-        size_t pairTotal  = 0;
-        uint8_t pairCount = 0;
-
-        extensionFieldSet.clusterID = cluster;
-        reader.Init(serialisedBytes);
-        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
-        ReturnErrorOnFailure(reader.EnterContainer(outer));
-        ReturnErrorOnFailure(reader.Next(
-            TLV::kTLVType_Array, TLV::ContextTag(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList)));
-        attributeValueList.Decode(reader);
-
-        // Verify size of list
-        ReturnErrorOnFailure(attributeValueList.ComputeSize(&pairTotal));
-        VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+        ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
 
+        uint8_t pairCount  = 0;
```

Comment:
Isn't _this_ where the count check against mAVPairs should happen, since this is where we care about it?